src/main/java/org/sonarlint/intellij/ui/ruledescription/RuleParsingUtils.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2023 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.ui.ruledescription
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.fileTypes.FileType
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.VerticalFlowLayout
+import com.intellij.openapi.util.Disposer
+import com.intellij.psi.XmlElementFactory
+import com.intellij.ui.ScrollPaneFactory
+import com.intellij.ui.components.JBPanel
+import org.apache.commons.lang.StringEscapeUtils
+import org.apache.commons.lang.StringUtils
+import org.sonarlint.intellij.ui.ruledescription.section.CodeExampleFragment
+import org.sonarlint.intellij.ui.ruledescription.section.CodeExampleType
+import org.sonarlint.intellij.ui.ruledescription.section.HtmlFragment
+import org.sonarlint.intellij.ui.ruledescription.section.Section
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import javax.swing.JScrollPane
+import javax.swing.ScrollPaneConstants
+
+class RuleParsingUtils {
+
+    companion object {
+        private const val PRE_TAG_ENDING = "</pre>"
+
+        fun parseCodeExamples(
+            project: Project, parent: Disposable, htmlDescription: String, fileType: FileType,
+        ): JScrollPane {
+            val mainPanel = JBPanel<JBPanel<*>>(VerticalFlowLayout(0, 0))
+            var remainingRuleDescription = htmlDescription
+            var computedRuleDescription = ""
+            var matcherStart: Matcher = Pattern.compile("<pre[^>]*>").matcher(remainingRuleDescription)
+            var matcherEnd: Matcher = Pattern.compile(PRE_TAG_ENDING).matcher(remainingRuleDescription)
+
+            val section = Section()
+            val xmlElementFactory = XmlElementFactory.getInstance(project)
+            while (matcherStart.find() && matcherEnd.find()) {
+                val front: String = remainingRuleDescription.substring(0, matcherStart.start()).trim()
+
+                if (front.isNotBlank()) {
```

Comment:
We are using similar logic on line 87. Maybe we can extract this as a function too.

Reply:
I think the benefits to extract this as a function are not great enough, it would make the code maybe less readable for minimal factorizing.